kernel/CMakeLists.txt

```diff
@@ -117,8 +117,35 @@ target_sources_ifdef(CONFIG_OBJ_CORE              kernel PRIVATE obj_core.c)
 
 if(${CONFIG_KERNEL_MEM_POOL})
   target_sources(kernel PRIVATE mempool.c)
-endif()
 
+  set(final_heap_size ${CONFIG_HEAP_MEM_POOL_SIZE})
+
+  # Import all custom HEAP_MEM_POOL minimum size requirements
+  import_kconfig(CONFIG_HEAP_MEM_POOL_MIN_SIZE_ ${DOTCONFIG} min_size_keys)
+
+  set(max_min_size 0)
+
+  foreach(min_size ${min_size_keys})
+    if("${${min_size}}" GREATER "${max_min_size}")
+      set(max_min_size "${${min_size}}")
+    endif()
+  endforeach()
+
+  if(CONFIG_HEAP_MEM_POOL_SIZE LESS "${max_min_size}")
+    if(NOT CONFIG_HEAP_MEM_POOL_IGNORE_MIN)
+      set(final_heap_size "${max_min_size}")
+    endif()
+    # Only warn if default value 0 has been modified
+    if(NOT CONFIG_HEAP_MEM_POOL_SIZE EQUAL 0)
```

Comment:
CONFIG_HEAP_MEM_POOL_SIZE can never be zero because this line evaluates to false when `CONFIG_HEAP_MEM_POOL_SIZE =0`:
https://github.com/zephyrproject-rtos/zephyr/blob/4a199d7f0fd43b5d18e21eec2f1651fad52e853c/kernel/CMakeLists.txt#L118

Reply:
scratch that, somehow I was reading this as: `if(${CONFIG_HEAP_MEM_POOL_SIZE})`